lib/serialize.js

```diff
@@ -0,0 +1,230 @@
+
+  // export as Multikey
+  const exported = {};
+  if(includeContext) {
+    exported['@context'] = MULTIKEY_CONTEXT_V1_URL;
+  }
+  exported.id = keyPair.id;
+  exported.type = 'Multikey';
+  exported.controller = keyPair.controller;
+
+  if(publicKey) {
+    // convert `x` coordinate to compressed public key
+    const x = base64url.decode(jwk.x);
+    const y = base64url.decode(jwk.y);
+    // public key size is always secret key size + 1
+    const publicKeySize = secretKeySize + 1;
+    // leave room for multicodec header (2 bytes)
+    const multikey = new Uint8Array(2 + publicKeySize);
+    setPublicKeyHeader({keyPair, buffer: multikey});
+    // use even / odd status of `y` coordinate for compressed header
+    const even = y[y.length - 1] % 2 === 0;
+    multikey[2] = even ? 2 : 3;
+    // write `x` coordinate at end of multikey buffer to zero-fill it
+    multikey.set(x, multikey.length - x.length);
+    exported.publicKeyMultibase = MULTIBASE_BASE58_HEADER +
+      base58.encode(multikey);
+  }
+
+  if(secretKey) {
+    const d = base64url.decode(jwk.d);
+    // leave room for multicodec header (2 bytes)
+    const multikey = new Uint8Array(2 + secretKeySize);
+    setSecretKeyHeader({keyPair, buffer: multikey});
+    // write `d` at end of multikey buffer to zero-fill it
+    multikey.set(d, multikey.length - d.length);
+    exported.secretKeyMultibase = MULTIBASE_BASE58_HEADER +
+      base58.encode(multikey);
+  }
+
+  return exported;
+}
+
+// imports key pair
+export async function importKeyPair({
+  id, controller, secretKeyMultibase, publicKeyMultibase
+}) {
+  if(!publicKeyMultibase) {
+    throw new TypeError('The "publicKeyMultibase" property is required.');
+  }
+
+  const keyPair = {id, controller};
+
+  // import public key
+  if(!(publicKeyMultibase && typeof publicKeyMultibase === 'string' &&
+    publicKeyMultibase[0] === MULTIBASE_BASE58_HEADER)) {
+    throw new TypeError(
+      '"publicKeyMultibase" must be a multibase, base58-encoded string.');
+  }
+  const publicMultikey = base58.decode(publicKeyMultibase.slice(1));
+
+  // set named curved based on multikey header
+  const algorithm = {
+    name: ALGORITHM,
+    namedCurve: getNamedCurve({publicMultikey})
+  };
+
+  // import public key; convert to `spki` format because `jwk` doesn't handle
+  // compressed public keys
+  const spki = _toSpki({publicMultikey});
+  keyPair.publicKey = await webcrypto.subtle.importKey(
+    'spki', spki, algorithm, EXTRACTABLE, ['verify']);
+
+  // import secret key if given
+  if(secretKeyMultibase) {
+    if(!(typeof secretKeyMultibase === 'string' &&
+    secretKeyMultibase[0] === MULTIBASE_BASE58_HEADER)) {
+      throw new TypeError(
+        '"secretKeyMultibase" must be a multibase, base58-encoded string.');
+    }
+    const secretMultikey = base58.decode(secretKeyMultibase.slice(1));
+
+    // FIXME: ensure secret key multikey header appropriately matches the
```

Comment:
@dlongley This doesn't make sense to me. What is a "multikey header"? Perhaps, this is in reference to the `publicKeyMultibase` header on the Multikey. Did you mean ensure that the `publicKeyMultibase` matches its expected corresponding  `secretKeyMultibase`?

@kezike We will need to address this in another PR before release.

Reply:
Created Issue #3 to address this `FIXME` comment